### PR TITLE
Fix #1554 Camera settings does not work

### LINKED ISSUE #1554: Camera settings does not work
**Actual Behaviour**

App stops working when pressed the *Manage camera settings*
**This error occurred in devices which supports "supports_camera2" bundle**

**LogCat for the issue**
```
E/AndroidRuntime: FATAL EXCEPTION: main
                                                                      Process: org.fossasia.phimpme, PID: 14811
                                                                      java.lang.NullPointerException: Attempt to invoke virtual method 'void android.preference.Preference.setOnPreferenceClickListener(android.preference.Preference$OnPreferenceClickListener)' on a null object reference
                                                                          at org.fossasia.phimpme.opencamera.Camera.MyPreferenceFragment.onCreate(MyPreferenceFragment.java:266)
                                                                          at android.app.Fragment.performCreate(Fragment.java:2489)
                                                                          at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1237)
                                                                          at android.app.FragmentTransition.addToFirstInLastOut(FragmentTransition.java:1319)
                                                                          at android.app.FragmentTransition.calculateFragments(FragmentTransition.java:1203)
                                                                          at android.app.FragmentTransition.startTransitions(FragmentTransition.java:101)
                                                                          at android.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2178)
                                                                          at android.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2136)
                                                                          at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2043)
                                                                          at android.app.FragmentManagerImpl$1.run(FragmentManager.java:719)
                                                                          at android.os.Handler.handleCallback(Handler.java:790)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                          at android.os.Looper.loop(Looper.java:164)
                                                                          at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                          at java.lang.reflect.Method.invoke(Native Method)
                                                                          at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

Error is "null object" on line 265 in MyPreferenceFragment.java
```
final Preference pref = findPreference("preference_use_camera2");
            pref.setOnPreferenceClickListener(new OnPreferenceClickListener()...
```

**Screenshots of the issue**

screenshot_20171227-015749

**Would you like to work on the issue?**

yes

### PR DESCRIPTION
**Pull request by ashley8jain**
Fix #1554

Changes: just added SwitchPreference named "preference_use_camera2" as key in preferences.xml

Screenshots for the change:
whatsapp image 2018-01-08 at 6 11 23 pm

### CODE REVIEW

File: app/src/main/res/xml/preferences.xml [hunk 1]
```diff
@@ -406,5 +406,20 @@
...
+        android:title="@string/preference_category_online"
+        >
+
+        <!--  if we move this to another PreferenceGroup, we should update code for the findPreference() call -->
+        <SwitchPreference
+            android:key="preference_use_camera2"
+            android:title="@string/preference_use_camera2"
+            android:summary="@string/preference_use_camera2_summary"
+            android:defaultValue="false"
+            />
```
Comment (mayank8318) [inline app/src/main/res/xml/preferences.xml]: Please follow proper formatting (look at other preferences and how the brackets are closed).

File: app/src/main/res/xml/preferences.xml [hunk 2]
```diff
@@ -209,6 +209,14 @@
                 android:title="@string/preference_keep_display_on" />
```
Comment (mayank8318) [inline app/src/main/res/xml/preferences.xml]: @ashley8jain I meant follow this pattern please :)

**Comment by mayank8318** [comment]
@ashley8jain Also do not press the UPDATE button on github. Do you know how to fetch from upstream and rebase? Feel free to ask if you need any help.

**Comment by ashley8jain** [comment]
@mayank8318 rebase done.Actually this is my first PR request.

**Comment by ashley8jain** [comment]
@mayank8318 now pattern changed. Please approve this

**Comment by ashley8jain** [comment]
@mayank8318 Please see the new changes. review this

**Comment by ashley8jain** [comment]
@mayank8318 Please see the new changes. review this

**Comment by ashley8jain** [comment]
@anantprsd5 @mohitmanuja merge this PR please

**Comment by ashley8jain** [comment]
@isuruAb merge this squashed commit please

**Pull request merged**